tests/e2e/core-tests/specs/shopper/front-end-single-product.test.js

```diff
@@ -67,6 +69,50 @@ const runSingleProductPageTest = () => {
 			await expect(page).toMatchElement('.cart-empty', {text: 'Your cart is currently empty.'});
 		});
 	});
+
+	describe('Grouped Product Page', () => {
+		beforeAll(async () => {
+			await merchant.login();
+			groupedPostIdValue = await createGroupedProduct();
+			await merchant.logout();
+		})
+
+		it('should be able to add grouped products to the cart', async () => {
+			// Add a grouped product to cart
+			await shopper.goToProduct(groupedPostIdValue);
+			await page.waitForSelector('form.grouped_form');
+			await shopper.addToCart();
+			await expect(page).toMatchElement('.woocommerce-error',
+			 {text: 'Please choose the quantity of items you wish to add to your cart…'});
+			const quantityFields = await page.$$('div.quantity input.qty');
+			await quantityFields[0].click({clickCount: 3});
+			await quantityFields[0].type('5');
+			await quantityFields[1].click({clickCount: 3});
+			await quantityFields[1].type('5');
+			await shopper.addToCart();
+			await expect(page).toMatchElement('.woocommerce-message',
+			 {text: '“'+simpleProductName+' 1” and “'+simpleProductName+' 2” have been added to your cart.'});
```

Comment:
Just as an alternative syntax if you're interested (nothing you need to change here), you can replace this string concatenation with `${}`, so this could become:

```suggestion
			 {text: `${simpleProductName} 1 and ${simpleProductName} 2 have been added to your cart.`}
```

If you use this syntax, just need to make sure you wrap the string in backticks (`) instead of regular single or double quotes.

Reply:
Interesting approach. I tried but it was failing. Will keep currently as-is and will try your approach in the future, looks nice.

Reply:
Ah too bad! As you said we can try the approach in the future instead :)